### PR TITLE
Feat: ORV2-1933 Cleanup PayBC Redirection Flow

### PR DESCRIPTION
**Pull request by erikataot**
# Description
- Updated the PayBC redirection flow to clean it up and remove the double `??` that was appearing

image

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist

- [ ] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have already been accepted and merged

## Further comments

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://onroutebc-1159-frontend.apps.silver.devops.gov.bc.ca)
- [Vehicles](https://onroutebc-1159-vehicles.apps.silver.devops.gov.bc.ca/api)
- [Dops](https://onroutebc-1159-dops.apps.silver.devops.gov.bc.ca/api)
- [TPS-Migration](https://onroutebc-1159-tps-migration.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/onroutebc/actions/workflows/analysis.yml)

After merge, new images are promoted to:
- [Merge Workflow](https://github.com/bcgov/onroutebc/actions/workflows/merge.yml)

### CODE REVIEW

File: frontend/src/common/types/paymentTransaction.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    pgCardType: string;
+    pgTransactionDate: string;
+    pgCvdId: number;
+    pgPaymentMethod: string;
+    pgMessageId: number;
+    pgMessageText: string;
+    transactionId: string;
+    totalTransactionAmount: number;
+    applicationDetails: Array<ApplicationDetails>;
+};
```
Comment (zgong-gov) [inline frontend/src/common/types/paymentTransaction.ts]: How do these types differ from the existing ones in payment.d.ts? Was there a reason why you created these additional types?

File: frontend/src/features/permits/helpers/payment.ts [hunk 2]
```diff
@@ -36,7 +35,10 @@ export const getPayBCPaymentDetails = (
     avsResult: getDefaultRequiredVal("", params.get("avsResult")),
     cardType: getDefaultRequiredVal("", params.get("cardType")),
     cvdId: applyWhenNotNullable((cvdId) => Number(cvdId), params.get("cvdId")),
-    trnApproved: trnApproved,
+    trnApproved: applyWhenNotNullable(
+      (approved) => Number(approved),
+      params.get("trnApproved"),
```
Comment (zgong-gov) [inline frontend/src/features/permits/helpers/payment.ts:39]: Perhaps `trnApproved: applyWhenNotNullable((approved) => Number(approved), params.get("trnApproved"), 0)`, with the 0 as the last param

File: frontend/src/features/permits/pages/Payment/PaymentRedirect.tsx [hunk 3]
```diff
@@ -66,23 +31,17 @@ export const PaymentRedirect = () => {
...
-
-  const path = getDefaultRequiredVal("", searchParams.get("path"));
-  const { permitIds, trnApproved } = parseRedirectUriPath(path);
-  const transactionQueryString = exportPathFromSearchParams(
-    searchParams,
-    trnApproved,
-  );
   const transactionId = getDefaultRequiredVal("", searchParams.get("ref2"));
-
+  const [applicationIds, setApplicationIds] = useState<string[] | []>([]);
```
Comment (zgong-gov) [inline frontend/src/features/permits/pages/Payment/PaymentRedirect.tsx]: Perhaps just `useState<string[]>([])` is enough?

File: frontend/src/features/permits/pages/Payment/PaymentRedirect.tsx [hunk 4]
```diff
@@ -97,24 +56,38 @@ export const PaymentRedirect = () => {
   }, [paymentDetails.trnApproved]);
 
   useEffect(() => {
-    if (issuedPermit.current === false) {
-      const permitIdsArray = getPermitIdsArray(permitIds);
+    const ids:string[] = [];
+    getPaymentByTransactionId(transactionId)
+      .then((response) => {
+        response?.applicationDetails?.forEach((application) => {
```
Comment (zgong-gov) [inline frontend/src/features/permits/pages/Payment/PaymentRedirect.tsx:62]: Shouldn't we use a useQuery for this?

File: frontend/src/common/types/paymentTransaction.ts (hunk 1 above)
Comment (erikataot) [reply frontend/src/common/types/paymentTransaction.ts]: Updated to use `StartTransactionResponseData` thanks

File: frontend/src/features/permits/pages/Payment/PaymentRedirect.tsx (hunk 3 above)
Comment (erikataot) [reply frontend/src/features/permits/pages/Payment/PaymentRedirect.tsx]: Yes, it's enough - thanks

File: frontend/src/features/permits/helpers/payment.ts (hunk 2 above)
Comment (erikataot) [reply frontend/src/features/permits/helpers/payment.ts:39]: good idea - thanks

File: frontend/src/features/permits/pages/Payment/PaymentRedirect.tsx [hunk 5]
```diff
@@ -97,24 +56,38 @@ export const PaymentRedirect = () => {
...
+        setApplicationIds(ids);
+      })
+
+  }, [transactionId]);
+
+  useEffect(() => {
+    if (applicationIds?.length > 0 && issuedPermit.current === false) {
 
-      if (permitIdsArray.length === 0) {
+      if (applicationIds?.length === 0) {
```
Comment (krishnan-aot) [inline frontend/src/features/permits/pages/Payment/PaymentRedirect.tsx:76]: Line 74 and Line 76 are contradictory if checks. applicationIds?.length > 0  and applicationIds?.length === 0

**Pull request closed without merging**